src/condition.cpp

```diff
@@ -691,6 +708,100 @@ bool ConditionAttributes::setParam(ConditionParam_t param, int32_t value)
 	}
 }
 
+int32_t ConditionAttributes::getParam(ConditionParam_t param)
+{
+	int32_t value = ConditionGeneric::getParam(param);
+	if (value != std::numeric_limits<int32_t>().max()) {
+		return value;
+	}
+
+	switch (param) {
+		case CONDITION_PARAM_SKILL_FIST:
+			return skills[SKILL_FIST];
+
+		case CONDITION_PARAM_SKILL_FISTPERCENT:
+			return skillsPercent[SKILL_FIST];
+
+		case CONDITION_PARAM_SKILL_CLUB:
+			return skills[SKILL_CLUB];
+
+		case CONDITION_PARAM_SKILL_CLUBPERCENT:
+			return skillsPercent[SKILL_CLUB];
+
+		case CONDITION_PARAM_SKILL_SWORD:
+			return skills[SKILL_SWORD];
+
+		case CONDITION_PARAM_SKILL_SWORDPERCENT:
+			return skillsPercent[SKILL_SWORD];
+
+		case CONDITION_PARAM_SKILL_AXE:
+			return skills[SKILL_AXE];
+
+		case CONDITION_PARAM_SKILL_AXEPERCENT:
+			return skillsPercent[SKILL_AXE];
+
+		case CONDITION_PARAM_SKILL_DISTANCE:
+			return skills[SKILL_DISTANCE];
+
+		case CONDITION_PARAM_SKILL_DISTANCEPERCENT:
+			return skillsPercent[SKILL_DISTANCE];
+
+		case CONDITION_PARAM_SKILL_SHIELD:
+			return skills[SKILL_SHIELD];
+
+		case CONDITION_PARAM_SKILL_SHIELDPERCENT:
+			return skillsPercent[SKILL_SHIELD];
+
+		case CONDITION_PARAM_SKILL_FISHING:
+			return skills[SKILL_FISHING];
+
+		case CONDITION_PARAM_SKILL_FISHINGPERCENT: 
+			return skillsPercent[SKILL_FISHING];
+
+		case CONDITION_PARAM_STAT_MAXHITPOINTS:
+			return stats[STAT_MAXHITPOINTS];
+
+		case CONDITION_PARAM_STAT_MAXMANAPOINTS:
+			return stats[STAT_MAXMANAPOINTS];
+
+		case CONDITION_PARAM_STAT_MAGICPOINTS:
+			return stats[STAT_MAGICPOINTS];
+
+		case CONDITION_PARAM_STAT_MAXHITPOINTSPERCENT:
+			return statsPercent[STAT_MAXHITPOINTS];
+
+		case CONDITION_PARAM_STAT_MAXMANAPOINTSPERCENT:
+			return statsPercent[STAT_MAXMANAPOINTS];
+
+		case CONDITION_PARAM_STAT_MAGICPOINTSPERCENT:
+			return statsPercent[STAT_MAGICPOINTS];
+
+		case CONDITION_PARAM_DISABLE_DEFENSE:
+			return disableDefense ? 1 : 0;
+
+		case CONDITION_PARAM_SPECIALSKILL_CRITICALHITCHANCE:
+			return specialSkills[SPECIALSKILL_CRITICALHITCHANCE];
+
+		case CONDITION_PARAM_SPECIALSKILL_CRITICALHITAMOUNT:
+			return specialSkills[SPECIALSKILL_CRITICALHITAMOUNT];
+
+		case CONDITION_PARAM_SPECIALSKILL_LIFELEECHCHANCE:
+			return specialSkills[SPECIALSKILL_LIFELEECHCHANCE];
+
+		case CONDITION_PARAM_SPECIALSKILL_LIFELEECHAMOUNT:
+			return specialSkills[SPECIALSKILL_LIFELEECHAMOUNT];
+
+		case CONDITION_PARAM_SPECIALSKILL_MANALEECHCHANCE:
+			return specialSkills[SPECIALSKILL_MANALEECHCHANCE];
+
+		case CONDITION_PARAM_SPECIALSKILL_MANALEECHAMOUNT:
+			return specialSkills[SPECIALSKILL_MANALEECHAMOUNT];
+
+		default:
+			return value;
```

Comment:
You should move the initial check here instead:

```
default:
    return ConditionGeneric::getParam(param);
```
So the branching is unnecessary and the result is the same.

Reply:
True, not sure why I did it this way. It was few months ago xD

Reply:
Long time since we talked about this, but here we go. It should be more reasonable now.